dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/startup/InitTableAlteror.java

```diff
@@ -177,6 +177,24 @@ public void execute()
         executeSql( "UPDATE categorycombo SET publicaccess = 'rw------' WHERE name = 'default' or code = 'default'" );
 
         executeSql( "UPDATE userroleauthorities SET authority = 'F_RELATIONSHIPTYPE_PUBLIC_ADD' WHERE authority = 'F_RELATIONSHIPTYPE_ADD'" );
+
+        //Fixing problem with missing columns and default values in trackedentitycomment table
+        executeSql( "ALTER TABLE trackedentitycomment ADD COLUMN IF NOT EXISTS uid varchar(11) unique;" );
+        executeSql( "ALTER TABLE trackedentitycomment ADD COLUMN IF NOT EXISTS created timestamp;" );
+        executeSql( "ALTER TABLE trackedentitycomment ADD COLUMN IF NOT EXISTS lastupdated timestamp;" );
+        executeSql( "ALTER TABLE trackedentitycomment ADD COLUMN IF NOT EXISTS code varchar(50) unique;" );
+        executeSql( "ALTER TABLE trackedentitycomment ADD COLUMN IF NOT EXISTS lastupdatedby integer;" );
+
+        executeSql( "UPDATE trackedentitycomment SET uid = generate_uid() WHERE uid IS NULL;" );
+        executeSql( "UPDATE trackedentitycomment SET created = now() WHERE created IS NULL;" );
+        executeSql( "UPDATE trackedentitycomment SET lastupdated = now() WHERE lastupdated IS NULL;" );
+
+        executeSql( "ALTER TABLE trackedentitycomment ALTER COLUMN uid SET NOT NULL;" );
+        executeSql( "ALTER TABLE trackedentitycomment ALTER COLUMN created SET NOT NULL;" );
+        executeSql( "ALTER TABLE trackedentitycomment ALTER COLUMN lastupdated SET NOT NULL;" );
+
+        executeSql( "ALTER TABLE trackedentitycomment DROP CONSTRAINT IF EXISTS fk_lastupdateby_userid;" );
+        executeSql( "ALTER TABLE trackedentitycomment ADD CONSTRAINT fk_lastupdateby_userid FOREIGN KEY (lastupdatedby) REFERENCES userinfo(userinfoid);" );
```

Comment:
I don't think we need to drop the constraint as the statements in table alteror will fail silently and not abort other statements.

Reply:
Fixed. Didn't know that it fails silently. In the end, not needed at all as the columns are mentioned directly in the .hbm file.